### PR TITLE
Fix the memory leak in worker cancel scene

### PR DESCRIPTION
**Pull request by jayzhenghan**
https://github.com/Alluxio/alluxio/issues/14466

### CODE REVIEW

**Comment by jayzhenghan** [comment]
@yuzhu @ggezer Help comment，thanks

**Comment by yuzhu** [comment]
@jayzhenghan Thanks for catching this issue! @ggezer proposed #14494 seems like an alternative simpler solution.  Please let us know if you have any questions.

**Review by yuzhu — APPROVED** [review]

**Comment by yuzhu** [comment]
alluxio-bot, merge this please

**Comment by Xenorith** [comment]
alluxio-bot, cherry-pick this to branch-2.7 please

File: core/server/worker/src/main/java/alluxio/worker/grpc/DataMessageServerStreamObserver.java [hunk 1]
```diff
@@ -51,7 +51,12 @@ public void onNext(DataMessage<T, DataBuffer> value) {
     if (buffer != null) {
       mBufferRepository.offerBuffer(buffer, value.getMessage());
     }
-    mObserver.onNext(value.getMessage());
+    try {
+      mObserver.onNext(value.getMessage());
+    } catch (Throwable e) {
+      mBufferRepository.pollBuffer(value.getMessage());
```
Comment (dbw9580) [inline core/server/worker/src/main/java/alluxio/worker/grpc/DataMessageServerStreamObserver.java:57]: Missing release? The buffer contained in the `DataBuffer` can be a Netty ByteBuf that is reference counted and needs explicit releasing. See `alluxio.network.protocol.databuffer.NettyDataBuffer`

```suggestion
      mBufferRepository.pollBuffer(value.getMessage()).release();
```

**Pull request merged**